packages/strapi-utils/lib/models.js

```diff
@@ -75,13 +75,23 @@ module.exports = {
         });
       });
     } else if (_.has(attribute, 'via') && _.has(attribute, 'collection')) {
-      const relatedAttribute = models[attribute.collection].attributes[attribute.via];
+      const model = models[attribute.collection];
+
+      if (!model) {
+        throw new Error(
+          `The provided 'collection' in the attribute \`${attributeName}\` in the model ${_.upperFirst(
+            modelName
+          )} doesn't exist`
```

Comment:
I don't think you can get the plugin from which `modelName` is from that function, at least from what I understand. You could get from the other ones because it was specified in the `attributes` variable.